### ISSUE TITLE
traceback: removing task from pool

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3424]
Traceback on restart, presumably resulting from an unexpected/uncontrolled shutdown where the suites' task pool has got out of sync with the db?

```
2019-10-25T11:47:07Z ERROR - 23450101T0000Z
        Traceback (most recent call last):
          File "/common/fcm/cylc-7.8.3/lib/cylc/scheduler.py", line 247, in start
            self.run()
          File "/common/fcm/cylc-7.8.3/lib/cylc/scheduler.py", line 1596, in run
            self.process_task_pool()
          File "/common/fcm/cylc-7.8.3/lib/cylc/scheduler.py", line 1241, in process_task_pool
            if meth():
          File "/common/fcm/cylc-7.8.3/lib/cylc/task_pool.py", line 1061, in remove_spent_tasks
            self.remove(itask)
          File "/common/fcm/cylc-7.8.3/lib/cylc/task_pool.py", line 511, in remove
            del self.pool[itask.point][itask.identity]
        KeyError: 23450101T0000Z
```

Cylc is failing to remove something which isn't there. This shouldn't happen but the error is safe to ignore:

```diff
        if itask.tdef.name in self.myq:  # A reload can remove a task
            del self.queues[self.myq[itask.tdef.name]][itask.identity]
-        del self.pool[itask.point][itask.identity]
+        try:
+            del self.pool[itask.point][itask.identity]
+        except KeyError:
+            LOG.error(f'Unable to remove {itask.identity} from the pool - not present.')
        if not self.pool[itask.point]:
            del self.pool[itask.point]
```

The remove case is safe to ignore but other cases are more deadly. Should we ignore the error or does this need a more generic solution?

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by hjoliver** [comment]
> presumably resulting from an unexpected/uncontrolled shutdown where the suites' task pool has got out of sync with the db?

That shouldn't possible, right?

**Comment by oliver-sanders** [comment]
Fair point.

Sadly I don't have any evidence to go back to. The user was restarting a suite they thought should have been running which had probably died unexpectedly.

The issue was repeatable and was eventually solved with a warm start.

**Comment by hjoliver** [comment]
Unfortunately it's hard to know what to do here, without some detail on what triggered the problem.  Maybe just put in the `try` block as you suggest, and keep an eye out for any further reports?

**Comment by hjoliver** [comment]
Closing as wontfix - not relevant to Cylc 8 as "remove spent tasks" is a spawn-on-submit thing.
